pagopa/io-app
# feat: [IOBP-9] Add IDPay transaction QRCode scan screen

## Short description
This PR adds the screen for the IDPay transaction QRCode scan.

## List of changes proposed in this pull request
- ts/features/idpay/payment/components/BarcodeScanner.tsx: 
   - Added `BarcodeScanner` component, whihc allows the scan of a barcode/QR Code
   - Added `useIOBarcodeScanner`, hook that allows to use the `BarcodeScanner` component with a configuration
- ts/features/idpay/payment/screens/IDPayPaymentCodeScanScreen.tsx:
   - Added `BarcodeScanner` to the view
   - Added permission views, which shows to the user the right prompt to grant camera permissions
 - Added required locale keys
 - Added required pictograms

## How to test
- Tests should solve correctly.
- With the `io-dev-server`, from the wallet section of the app navigate to details screen of a discount type initiative (marked with a **[D]** in its name). From here, click on the payment authorization button and scan [this QRCode](https://continua.io.pagopa.it/qrcode.png?feat=idpay&trxcode=00000000).
